File: src/shared/components.types.ts
export interface ArchetypeComponent {
    type: string;
    faction: string;
    mass: number;
}

export interface VelocityComponent {
    value: Vector3;
    angular: Vector3;
}

export interface CFrameComponent {
    value: CFrame;
    lastUpdated: number;
}

export interface WeaponStateComponent {
    isCharging: boolean;
    nextTimer: number;
    ammoCapacity: number;
}

export interface HealthComponent {
    current: number;
    max: number;
    isInvulnerable: boolean;
}

export interface ExplosionTriggerComponent {
    radius: number;
    force: number;
    active: boolean;
}

export interface DamageComponent {
    value: number;
}

export interface DamagePayloadComponent {
    value: number;
    targetEntityId: number;
}

File: .aura/services/shells/immutable-2/components.types.jl
AuraComponentPassport(
    id = "GalaxyArchetype_v38_5",
    status = "active",
    version = 2,
    vocabularyContract = "Aura_Galaxy_Vocabulary_v7.0",
    flameworkPattern = "Component",
    className = "GalaxyComponents",
    rojoTarget = "src/shared/components.types.ts",
    context = "Polny semanticheskiy pasport komponentov Matter ECS dlya kosmicheskogo simulatora",
    
    # Component specification (Flat structure)
    components = Dict(
        "ArchetypeComponent" => Dict(
            "type" => "string",
            "faction" => "string",
            "mass" => "number"
        ),
        "VelocityComponent" => Dict(
            "value" => "Vector3",
            "angular" => "Vector3"
        ),
        "CFrameComponent" => Dict(
            "value" => "CFrame",
            "lastUpdated" => "number"
        ),
        "WeaponStateComponent" => Dict(
            "isCharging" => "boolean",
            "nextTimer" => "number",
            "ammoCapacity" => "number"
        ),
        "HealthComponent" => Dict(
            "current" => "number",
            "max" => "number",
            "isInvulnerable" => "boolean"
        ),
        "ExplosionTriggerComponent" => Dict(
            "radius" => "number",
            "force" => "number",
            "active" => "boolean"
        ),
        "DamageComponent" => Dict(
            "value" => "number"
        ),
        "DamagePayloadComponent" => Dict(
            "value" => "number",
            "targetEntityId" => "number"
        )
    )
)

# AURA_END
